### ISSUE TITLE
I added support of Tensor Cores in my fork (Detection 3x times and Training 2x times)

### ISSUE THREAD

**Issue by AlexeyAB** [pjreddie/darknet#704]
I added support of Tensor Cores in my fork: https://github.com/AlexeyAB/darknet
To accelerate **Detection 3x** times and **Training 2x** times (with the same Loss and mAP) using Tensor Cores are required:
* nVidia GPU **Volta**: Tesla V100, Titan V, Quadro GV100, DGX-2
* CUDA >= 9.0 and cuDNN >= 7.0 

* **on Windows**:  open `\darknet.sln` -> (right click on project) -> properties -> C/C++ -> Preprocessor -> Preprocessor Definitions, and add at the beginning of line: `CUDNN_HALF;`

* **on Linux**: open `Makefile` and set `CUDNN_HALF=1`: https://github.com/AlexeyAB/darknet/blob/140333977cea0ba9e384cd38fd01013a8915ef60/Makefile#L3

Desirable to use OpenCV 3.4.0 that is used for 3x times accelerated data augmentation.

----

`GPU=1 CUDNN=1 CUDNN_HALF=1 OPENCV=1` in the Makefile 

* **Detection** - Forward inference time:

| Model | FP32 (Tesla V100), sec | Tensor Cores FP16/32 (Tesla V100), sec | Speedup X times|
|---|---|---|---|
| yolov3.cfg | 0.031 | 0.011 | **2.8x** |
| yolo-voc.2.0.cfg | 0.02 | 0.0062 | **3.2x** |
| tiny-yolo-voc.cfg | 0.003 | 0.0027 | **1.1x** |

----

* **Training** - Forward+Backward+Update time (`batch=64 subdivision=4 width=416 height=416 random=0`):

| Model | FP32 (Tesla V100), sec | Tensor Cores FP16/32 (Tesla V100), sec | Speedup X times|
|---|---|---|---|
| yolov3.cfg | 1.9 | 1.27 | **1.5x** |
| yolo-voc.2.0.cfg | 0.89 | 0.39 | **2.3x** |
| tiny-yolo-voc.cfg | 0.23 | 0.165 | **1.4x** |

**Comment by Iamthegeekdavid** [comment]
@AlexeyAB , do you expect any performance improvement of CUDNN_HALF for the Nvidida Pascal Cards (e.g., Geforce 1080 ti). Thanks,

**Comment by AlexeyAB** [comment]
@Iamthegeekdavid No. It is only for Volta, Turing and later more modern GPUs

**Comment by AlexeyAB** [comment]
@Iamthegeekdavid For Pascal GPU you can try to use this repo  https://github.com/AlexeyAB/yolo2_light and use flag`-quantized` at the end of command. You will get ~+30% speedup with ~-1% mAP on default yolov2/yolov3 models by using INT8 quantization on DP4A Pascal GPU. More: https://github.com/AlexeyAB/darknet/issues/1417

**Comment by wesolost** [comment]
@AlexeyAB, I check the code, It seem that, you only use the HALF_PRESICION when calling the FWD and BWD API in CUDA CUDNN, and then tranfers the data to SINGLE_PERSICION. So is there any problem here? I test the classifier with resnet152( Tesla V100), and It was even worse with CUNDD_HALF=1, 0.008s(CUNDNN_HALF = 0) VS 0.010s(CUNDD_HALF=1).

**Comment by AlexeyAB** [comment]
@wesolost Yes, it adds some overhead, but even in this case - it gives 3x times speedup on Yolo v3 and v2. 
Tranfer the data by using SINGLE_PERSICION allows us to use any default layers and use any future layers without any changes.

**Comment by BotCyborg** [comment]
Has anyone with access to RTX 2080 or 2080 TI tested what kind of speedup is possible with CUDNN_HALF?

**Comment by Matcoil** [comment]
is your solution suitable for RTX 2080ti, RTX 2080 or RTX 2070 graphic cards? (just double checking, since there are too much money involved :-) ).

**Comment by AlexeyAB** [comment]
@Matcoil Yes, you can use RTX 2080ti on Tensor Cores for FP16/FP32 and may be for INT8: https://github.com/AlexeyAB/darknet/issues/1822#issuecomment-433251255

**Comment by joeythegod** [comment]
@AlexeyAB Sorry for disturbing, I changed my GPU from GTX1070 to RTX2080, it can compile YoloV3 successfully but the demo prediction goes wrong. Is there any suggestion? Can't find a solution on the internet :(

**Comment by BotCyborg** [comment]
> @AlexeyAB Sorry for disturbing, I changed my GPU from GTX1070 to RTX2080, it can compile YoloV3 successfully but the demo prediction goes wrong. Is there any suggestion? Can't find a solution on the internet :(

Not sure, if you have the same issue, but I had something similar after upgrading from GTX 1080TI to RTX 2080TI. I had updated cuda 9.1 to 10 but didn't update cudnn. Issue was resolved after updating cudnn to the appropriate version. Hope this helps!

**Comment by AlexeyAB** [comment]
@joeythegod 

* Check that you use cuDNN version that correspnods to your CUDA version, as said @BotCyborg 

* If it doesn't help, then try to set `CUDNN_HALF=0`, does it help?

* If it doesn't help, then try to set `CUDNN=0`, does it help?

* Also try to un-comment this line: https://github.com/AlexeyAB/darknet/blob/9f7d7c58b5284eec9a1b96eb30c63ae254ded740/Makefile#L27

**Comment by KireinaHoro** [comment]
Can you try to merge the master branch back into your fork?  The API for darknet has changed greatly since you forked it, and your fork is missing various headers (notably `include/darknet.h`) to function properly for programs that use darknet.  Thanks!

**Comment by KireinaHoro** [comment]
Would you please merge the HEAD of pjreddie/darknet into your branch?  A lot of changes have taken place, and I'm interested in testing my own program (written upon the user API header `include/darknet.h`).  Thanks!

**Comment by shenyingying** [comment]
> Has anyone with access to RTX 2080 or 2080 TI tested what kind of speedup is possible with CUDNN_HALF?

when i compile yolo ,I have something wrong. the envirment is RTX2080, cuda 9.0,cudnn 7.0,the result in  gpu and cpu is different,there is no output in gpu compile.can anyone help me

**Comment by AlexeyAB** [comment]
@shenyingying 

Install CUDA 10.0 and cuDNN v7.4.2 (Dec 14, 2018), for CUDA 10.0.

I successfully use https://github.com/AlexeyAB/darknet with `GPU=1 CUDNN=1 CUDNN_HALF=1` in the `Makefile` on RTX 2070 with Tensor Cores.

**Comment by AlexeyAB** [comment]
@shenyingying 

I tested on:

GPU nVidia GeForce RTX 2070 with Tensor Cores:

* **Windows** 7 x64 (or **Linux** Debian 8.2 x86_64)
* **MSVS** 2015 (or **GCC** 4.9.2)
* CUDA 10.0
* cuDNN v7.4.2 (Dec 14, 2018), for CUDA 10.0
* OpenCV 3.3.0

**Comment by jasonabsher1** [comment]
@AlexeyAB, I'm curious if you could describe how you were able to get CUDA 10.0 to work with TensorFlow-GPU? I tried this, but it seems that they're incompatible. While trying to run "import tensorflow as tf" in python, I recieved the following error: libcublas.so.9.0: cannot open shared object file: no such file or directory. I noticed elsewhere that others have had issues and suggest that CUDA 9.0 is necessary for use with tensor flow. I'm running Ubuntu 18.04, and an RTX 2070 GPU. Any ideas how I can get this running on my my machine? - I would much prefer to use the newest CUDA release. Thanks in advance for any help!

**Comment by shenyingying** [comment]
Ubuntu18.04，CUDA10,  tensorflow-gpu the highest  cuda9.0,it's need gcc degrade发自我的华为手机-------- 原始邮件 --------主题：Re: [pjreddie/darknet] I added support of Tensor Cores in my fork (Detection 3x times and Training 2x times) (#704)发件人：jasonabsher1 收件人：pjreddie/darknet 抄送：BlueSandals ,Mention @AlexeyAB, I'm curious if you could describe how you were able to get CUDA 10.0 to work with TensorFlow-GPU? I tried this, but it seems that they're incompatible. While trying to run "import tensorflow as tf" in python, I recieved the following error: libcublas.so.9.0: cannot open shared object file: no such file or directory. I noticed elsewhere that others have had issues and suggest that CUDA 9.0 is necessary for use with tensor flow. I'm running Ubuntu 18.04, and an RTX 2070 GPU. Any ideas how I can get this running on my my machine? - I would much prefer to use the newest CUDA release. Thanks in advance for any help!

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/pjreddie/darknet","title":"pjreddie/darknet","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/pjreddie/darknet"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jasonabsher1 in #704: @AlexeyAB, I'm curious if you could describe how you were able to get CUDA 10.0 to work with TensorFlow-GPU? I tried this, but it seems that they're incompatible. While trying to run \"import tensorflow as tf\" in python, I recieved the following error: libcublas.so.9.0: cannot open shared object file: no such file or directory. I noticed elsewhere that others have had issues and suggest that CUDA 9.0 is necessary for use with tensor flow. I'm running Ubuntu 18.04, and an RTX 2070 GPU. Any ideas how I can get this running on my my machine? - I would much prefer to use the newest CUDA release. Thanks in advance for any help!"}],"action":{"name":"View Issue","url":"https://github.com/pjreddie/darknet/issues/704#issuecomment-450615449"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/pjreddie/darknet/issues/704#issuecomment-450615449",
"url": "https://github.com/pjreddie/darknet/issues/704#issuecomment-450615449",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by tmvlemk** [comment]
@AlexeyAB Hello, AlexeyAB. Thank you for your contributions!! According to NVIDIA guides for tensor cores, it seems that we can get much faster performance than you did (training 2x, detection 3x) Using mixed-precision for 32-bit floating point operations, V100 accelerator a peak throughout 12 times of the previous generation like P100. Your support for tensor cores is fully optimized like that? or is there any room for optimizations? Thanks for your help.

**Comment by AlexeyAB** [comment]
@tmvlemk Hi,

* V100 (enabled Tensor Cores) faster than P100 12x times only theoretically.
* V100 (enabled Tensor Cores) faster than V100 (disabled Tensor Cores) 8x times only theoretically.

In the real application V100 (enabled Tensor Cores) faster than V100 (disabled Tensor Cores) only ~2-4x times.

nVidia says that V100 **only 3x faster** than P100, `15.5 / 5.1 ~= 3x`: https://www.nvidia.com/en-us/data-center/tensorcore/
![data-center-tesla-v100-training-performance-chart-4-18-update-625-ud](https://user-images.githubusercontent.com/4096485/53355381-ee925f80-3939-11e9-96b7-fdb508398274.png)

But I compare V100 (enabled Tensor Cores) vs V100 (disabled Tensor Cores) **not with slow P100**. If I compared with P100, I would get more acceleration.

Also I added several optimizations, so currently I can get more acceleration on V100.

But there is still room for optimization.

----

TensorRT can achive only 3.5x acceleration on large batch and using FP16 (lower precision) instead of Mixed-precision FP16+FP32: https://medium.com/syncedreview/nvidias-volta-a-game-changer-for-ai-7a48e8d968dd

![0_7ukmdkr0qirw4mda_](https://user-images.githubusercontent.com/4096485/53356197-e76c5100-393b-11e9-98fc-cbdc9961a41d.png)

**Comment by tmvlemk** [comment]
@AlexeyAB Thank you for your prompt reply. 

I had used Titan Xp to train darknet before, and this time I'm trying to do that using new V100.
As you know, Titan Xp only supports 32-bit operation and it has about 12TFlops FP32 comparing 15TFlops of V100. So, simply we can expect 25% speed-up arithmetically for FP32 operation. Plus, if we use FP16(not mixed-precision), it's doubled and overall is about 2.5 times. 

Actually, I've got 2.15 times speed-up when I used V100 for training with CUDNN_HALF option. But, what I expected is much more speed improvements using tensor cores comparing with Titan Xp. How much speed-up do you expect when we're changing to V100 from Titan Xp?

**Comment by AlexeyAB** [comment]
@tmvlemk 

Since: https://en.wikipedia.org/wiki/List_of_Nvidia_graphics_processing_units
* Tesla P100 - 9.5 Tflops-32bit
* Titan Xp - 11 Tflops-32bit
* Tesla V100 - 14 Tflops-32bit (130 Tflops Tensor Cores)

I except acceleraton from Titan Xp to Tesla V100 ~=2x, i.e. less than from P100 to V100 ~= 3x - we can excpect the same acceleration as nVidia achived `3x` from P100 to V100.

But if you use Tesla V100 (**32 GB** not 16) or Quadro RTX 8000 (**48 GB**) 
with high `mini_batch = batch / subdivision = 32` or 64, f.e. `batch=64 subdivisions=1` then you can achive **4x-8x acceleration.**

There are several bottlenecks for full utilization of Tensor Cores:
* small mini-batch-size
* CPU-bottleneck in `[yolo]`-layers - I will try to solve it
* 1st and last layers (`channels % 8 != 0 || num_filters % 8 != 0`)  can't use Tensor Cores
* high number of small CUDA-calls
* slow data augmentation or slow HDD - you should compile with `OPENCV=1` in the Makefile, and try to increase this value, if your train not on Amazon EC2: https://github.com/AlexeyAB/darknet/blob/9b09abe12227a61c17c7b3c30c04642af50aa607/src/detector.c#L170

----

* During training the Tensor Cores are disabled for the first 3000 iterations - to achive good accuracy. For detection Tensor Cores are enabled always when CUDNN_HALF=1.
* Do you use the latest version of this repository? https://github.com/AlexeyAB/darknet
* What model do use use? Is it yolov3.cfg?
* What batch and subdivisions do you use?
* Did you compile Darknet with `GPU=1 CUDNN=1 CUDNN_HALF=1 OPENCV=1`? 
    (OPENCV=1 is necessary for fast data augmentation)

**Comment by tmvlemk** [comment]
@AlexeyAB 

Yes, I'm using your latest version of [https://github.com/AlexeyAB/darknet](url) and GV100 (32GB), so I will try your recommendation (high mini-batch)

But except for speed-up of training, I don't still understand poor accuracy for detection. mAP is decreased mid of 70% to 30% for half-precision training.

Here's my setting :
- darknet compiled with GPU=1 CUDNN=1 CUDNN_HALF=1 OPENCV=1 (3.2.0, 3.4.1)
- yolov3.cfg
- CUDA 10.0.130, cuDNN 7.4.2
- batch=64, subdivisions=16

Thanks!

**Comment by AlexeyAB** [comment]
@tmvlemk 

* To get good speedup you should use `batch=64 subdivision=4` or `batch=64 subdivision=2` or `batch=64 subdivision=1`

* Do you train on MS COCO dataset?

* Can you show two mAP training charts with CUDNN_HALF=1 and with CUDNN_HALF=0 ?

**Comment by tmvlemk** [comment]
@AlexeyAB 

I've tried to train yolov3 as your guide, but I couldn't set the subdivision below 8 because lower values (4 or 2 or 1) caused memory alloc error. 

I train on my own dataset, not COCO. The size of training data is about 7~8000.

With batch=64, subdivision=8, I could get slightly improvements about 10~20% more speed-up than before.

For the performance of FP16 training, the graph is like below.
<img src="https://user-images.githubusercontent.com/6690475/54007870-5604a680-41a7-11e9-9c9b-1e9f856460b4.png" width=500 />

and FP32 training result is here. Currently I don't have a mAP & loss chart of your version, so I attached my graph and mAP table. But there's no big difference. 

<img src="https://user-images.githubusercontent.com/6690475/54007655-315bff00-41a6-11e9-9ec8-55bba196e6bb.png" width=500 />

![image](https://user-images.githubusercontent.com/6690475/54008100-7da83e80-41a8-11e9-8114-a67737dcfe32.png)

**Comment by AlexeyAB** [comment]
@tmvlemk 

> and FP32 training result is here. Currently I don't have a mAP & loss chart of your version, so I attached my graph and mAP table. But there's no big difference.

* What repository did you use for FP32 training?
* What software did you use for mAP calculation for FP32 training?
* Do all your Training images have different size, or the same size?
* Do you use separate validation dataset for mAP calculation?
* Try to validate your FP32.weights file by using my repository https://github.com/AlexeyAB/darknet and such command `./darknet detector map data/obj.data yolo-obj.cfg backup\yolo-obj_250000.weights` and can you post screenshot of output

**Comment by tmvlemk** [comment]
@AlexeyAB 

> What repository did you use for FP32 training?

I used your repo (https://github.com/AlexeyAB/darknet) for FP32 training. But it's not the latest version when I trained it, so I couldn't get a chart graph like FP16 training and I saved every loss value of iteration and drew it. 
> What software did you use for mAP calculation for FP32 training?

mAP is also calculated using same source code that is used for FP32 training. That code was what you provided.
> Do all your Training images have different size, or the same size?

All training data are the same size.
>Do you use separate validation dataset for mAP calculation?

Yes,  I used separate valid dataset for mAP.
> Try to validate your FP32.weights file by using my repository https://github.com/AlexeyAB/darknet and such command ./darknet detector map data/obj.data yolo-obj.cfg backup\yolo-obj_250000.weights and can you post screenshot of output

Here's my screenshot.
<img src="https://user-images.githubusercontent.com/6690475/54026575-aa2c7c80-41e1-11e9-9362-726ac8644807.png" width=500 />

**Comment by AlexeyAB** [comment]
@tmvlemk Thanks!

> I used your repo (https://github.com/AlexeyAB/darknet) for FP32 training. But it's not the latest version when I trained it

Did you use the **latest** version of my repository for FP16 training? (on what date?)

**Comment by ttdd11** [comment]
@AlexeyAB Do we need to retrain using CUDNN_HALF=1 or can we just recompile using that flag and link to the yolo_cpp.dll? Also, is this flag backwards compatible? Ie. If we use the flag will detection on older cards (1080ti) still work just with no speed improvement. This repo is amazing thanks for all your hard work.

**Comment by AlexeyAB** [comment]
@ttdd11 

>  Do we need to retrain using CUDNN_HALF=1 or can we just recompile using that flag and link to the yolo_cpp.dll? 

You do not need to retrain. Just compile with `CUDNN_HALF=1` and use any old or new cfg/weights files.


> Also, is this flag backwards compatible? Ie. If we use the flag will detection on older cards (1080ti) still work just with no speed improvement. 

On the last version of https://github.com/AlexeyAB/darknet you can use any GPU with `CUDNN_HALF=1`. Detection on old GPUs (1080ti) will still work without speed improvement.

**Comment by tmvlemk** [comment]
> @tmvlemk Thanks!
> 
> > I used your repo (https://github.com/AlexeyAB/darknet) for FP32 training. But it's not the latest version when I trained it
> 
> Did you use the **latest** version of my repository for FP16 training? (on what date?)

@AlexeyAB 
Yes!

- Last commit info
commit f37ea75b28f01a87f08cb3cdea2f3d58f72224f8
Merge: cccc90b fd5312f
Author: Alexey <AlexeyAB@users.noreply.github.com>
Date:   Fri Feb 22 20:08:40 2019 +0300

    Merge pull request #2461 from cenit/dev/cenit/cmake_quickfix
    
    [CMakeLists.txt] fallback not enabled on windows

**Comment by SamNew1** [comment]
@AlexeyAB , have you ever tested processing a video using yolov3.weights on GPU nVidia GeForce RTX 2070 with Tensor Cores? How much is the FPS? Has the tensor core improved the speed a lot comparing with GTX 1080? Looking forward for your reply.

**Comment by ghsss** [comment]
For GTX 1660, which comes without tensor cores, but with some fp16 units, this implementation is viable?  Thanks!

https://www.techpowerup.com/gpu-specs/geforce-gtx-1660.c3365

**Comment by aimaicai** [comment]
Hi, what about 2060 RTX? I've tried to enable CUDNN_HALF=1, but the result was a light speed increase (~= +13%). On the other hand, the detections were very inaccurate for the standard yolo3 weights, and were disappearing from my own custom weights. Without the "CUDNN_HALF" it work perfectly (but a little slow for my purpose). Could it be possible? Have you some hint for helping me to go better than this?
Thank you very much

**Comment by aimaicai** [comment]
testing the classic dog.jpg against yolov3.cfg (236 MB COCO Yolo v3) I obtain "Predicted in 25.434000 milli-seconds" without CUDNN_HALF and "Predicted in 17.773000 milli-seconds" with CUDNN_HALF so more or less 30% faster (the detections are the same in this case). Is it the max that I can obtain from the card?
Thanks

**Comment by AlexeyAB** [comment]
@aimaicai 
On RTX 2070 and model `yolov3-spp.cfg` (608x608) you can get 
* 40.9ms with CUDNN_HALF=0 
* 27.2ms with CUDNN_HALF=1 (1.5x)

**Comment by aimaicai** [comment]
Thanks @AlexeyAB ! I test the spp cfg and weights and obtain from my 2060
-    49.8ms with CUDNN_HALF=0
-    31.6.ms with CUDNN_HALF=1 (1.58x)

So the gain is more or less the same for these cards. Maybe the 3x is for the top class cards. Thank you for your help, bye

**Comment by kingheungsun** [comment]
hi i'm using your custom darknet very well thx!!
i have some question to using that for my lab
i want to use pure darknet+using tensorcore 
so can i get where is chainging code for using tensorcore??

**Comment by azuryl** [comment]
I use TITAN Xp 

CUDNN_HALF=1  run darknet.py
predict use 0.03028s
CUDNN_HALF=0
predict use 0.03631s

**Comment by AlexeyAB** [comment]
@azuryl  TITAN Xp doesn't have Tensor Cores

**Comment by atagulmert** [comment]
@AlexeyAB Hi Alexey, thank you for your work. I'm trying to make use of tensor cores to improve performance on my RTX 2080 Ti. Instead of speedup, I'm achieving slower predictions with CUDNN_HALF = 1. E.g. coco.data, yolov3.cfg, yolov3.weights, data/dog.jpg  
CUDNN_HALF = 0 -> 12.76ms
CUDNN_HALF = 1 -> 16.95ms

My makefile is like that;
`GPU=1
CUDNN=1
CUDNN_HALF=1
OPENCV=1
AVX=1
OPENMP=1
LIBSO=0
ZED_CAMERA=0

ARCH= -gencode arch=compute_75,code=[sm_75,compute_75]
`

CUDA version 10.2. I at least expected it to achieve performance at same level as without tensor cores. It is achieving worse. Is there anything I should change within source code to make it compatible with architecture of my GPU or am I missing something else here?

**Comment by AlexeyAB** [comment]
@atagulmert I fixed it.

**Comment by ericluque** [comment]
@AlexeyAB 

I recently bought a Dell G3 i7 9gen with GTX 1660 ti. Increased RAM to 32 GB and sofar, Iḿ pretty happy :). I was trying to train Yolo V4 and did it very well with 416X416. Tried to train on 608 and 512, but it is exploding memory. I did my reasearch and changed the subdivisions and batches in a lot of ways (until subdivisions 256 indeed) . couldnt make it works. How can I discover how much memory it is using ? I would like to increase the size to gain a little bit more in precision.

Tx in advanced

**Comment by julia65203** [comment]
@AlexeyAB ,  do you expect any performance improvement of CUDNN_HALF=1 for the Nvidia Tesla T4. I tried it and it seems no big difference in the training time. BTW, Tesla T4 is the default configuration for Amazon deep learning AMI g4dn.xlarge. Thanks in advance!

**Comment by AlexeyAB** [comment]
Add in cfg, compile Darknet with CUDNN_HALF=1 and run training:
```
[net]
loss_scale=128.0
```

After 3000 iterations training will be accelerated 2x-3x depend on mini_batch size

**Comment by julia65203** [comment]
@AlexeyAB , thanks for your quick response! I have a couple of follow-up questions:

1) I based my config file from yolov4-custom.cfg. Per your suggestion, I will add the line of 'loss_scale=128.0'. Could you explain what this line does? Would this change impact the Loss and mAP? 

2) "After 3000 iterations training will be accelerated 2x-3x depend on mini_batch size"
Do you expect more acceleration if using a larger or smaller mini_batch size?

3) I used 'subdivisions=12' (out of memory if using 8). Would it work with CUDNN_HALF=1?

**Comment by AlexeyAB** [comment]
1. It shouldn't change visible loss

2. higher minibatch (batch/subdivisions) - higher speed

3. I don't know

**Comment by julia65203** [comment]
Thanks, @AlexeyAB . I will give it a try.

**Comment by ericluque** [comment]
@AlexeyAB 

Iḿ running my custom dataset and it is going good so far, unless for one reason. It is not burning in the 1k epochs files, just updating the *_last.weight file.

It was working on my old G3 and it is not working anymore. Any Known issue ? 

Running Ubuntu 20.04, CUDA 10.1 and CUDNN 8.0.5

Tx in advanced
